src/baseCommand.ts

```diff
@@ -342,9 +342,14 @@ export abstract class BaseCommand extends Command {
         return { first: passphrase, second: secondPassphrase };
     }
 
-    public getServiceProviderUNID(): string {
-        // returns the UNID of UNS forge factory
-        // will be configurable in case of multiple service providers available on UNS network
+    /**
+     * Return the UNID of this service provider depending on command flags
+     */
+    public getServiceProviderUNID(flags: Record<string, any>): string {
+        const urlCheckers: string[] = Managers.configManager.get("network.urlCheckers") || [];
+        if (flags.type === "url" && urlCheckers.length) {
+            return this.unsClientWrapper.network.milestone.urlCheckers[0];
+        }
```

Comment:
Why not use `urlCheckers`. What's the difference between `Managers.configManager.get("network.urlCheckers")` and `this.unsClientWrapper.network.milestone.urlCheckers`?

Reply:
this is a mistake. unsClientWrapper.network.milestone does not even exists